spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubProperties.java

```diff
@@ -50,6 +52,47 @@
 	 */
 	private String[] trustedPackages;
 
+	/**
+	 * The optional pull endpoint setting for the subscriber factory.
+	 */
+	private String subscriberPullEndpoint;
```

Comment:
Would it make sense to aggregate all of the `subscriber*` properties into its own class?

Reply:
Yes can do. I would prefer to make this an inner class named "Subscriber" rather than a separate class and then annotated with `@NestedConfigurationProperty`. Lemme know where I'm wrong, but as far as I know: the reason is we would prefer to use spring.cloud.gcp.pubsub.subscriber.blah, which would mean we would need a Subscriber.java class, which sounds like a misleading name for a java file and class if all it held were some settings. Therefore, I think it makes sense to have an inner class called GcpPubSubProperties.Subscriber. @meltsufin what do you think about that

Reply:
That's what we do in other places in Spring Boot: 

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

Although you don't need `@NestedConfigurationProperty` is the class is really nested. See those samples.

Reply:
Inner class should be fine.

Reply:
@meltsufin I also moved the publisher setting to its own class. PTAL